build.sbt

```diff
@@ -205,75 +221,131 @@ lazy val `kpl-ciris` = project
     Test / buildInfoOptions += BuildInfoOption.ConstantValue
   )
+  .jvmPlatform(allScalaVersions)
   .enableIntegrationTests
   .dependsOn(kpl, `shared-ciris`)
 
-lazy val `kpl-logging-circe` = project
+lazy val `kpl-logging-circe` = projectMatrix
   .settings(
     description := "JSON structured logging instances for the KPL, via Circe"
   )
+  .jvmPlatform(allScalaVersions)
   .enableIntegrationTests
   .dependsOn(kpl, `shared-circe`)
 
-lazy val `kpl-localstack` = project
+lazy val `kpl-localstack` = projectMatrix
   .settings(
     description := "A test-kit for working with Kinesis and Localstack, via the KPL"
   )
+  .jvmPlatform(allScalaVersions)
   .enableIntegrationTests
   .dependsOn(`aws-v1-localstack`, kpl)
 
-lazy val `kpl-tests` = project
+lazy val `kpl-tests` = projectMatrix
   .enablePlugins(NoPublishPlugin)
   .settings(
     description := "Integration Tests for the KPL"
   )
+  .jvmPlatform(allScalaVersions)
   .enableIntegrationTests
   .dependsOn(
     `kpl-localstack` % IT,
     `kpl-logging-circe` % IT
   )
 
-lazy val `kinesis-client` = project
+lazy val `kinesis-client` = projectMatrix
   .settings(
     description := "Cats tooling for the Java Kinesis Client",
     libraryDependencies ++= Seq(
       Aws.V2.kinesis,
       Log4Cats.slf4j
     )
   )
+  .jvmPlatform(allScalaVersions)
   .enableIntegrationTests
   .dependsOn(shared)
 
-lazy val `kinesis-client-logging-circe` = project
+lazy val `kinesis-client-logging-circe` = projectMatrix
   .settings(
     description := "JSON structured logging instances for the Java Kinesis Client, via Circe"
   )
+  .jvmPlatform(allScalaVersions)
   .enableIntegrationTests
   .dependsOn(
     `kinesis-client`,
     `shared-circe`,
     `aws-v2-localstack` % IT
   )
 
-lazy val `kinesis-client-localstack` = project
+lazy val `kinesis-client-localstack` = projectMatrix
   .settings(
     description := "A test-kit for working with Kinesis and Localstack, via the Kinesis Client project"
   )
+  .jvmPlatform(allScalaVersions)
   .enableIntegrationTests
   .dependsOn(`aws-v2-localstack`, `kinesis-client`)
 
-lazy val `kinesis-client-tests` = project
+lazy val `kinesis-client-tests` = projectMatrix
   .enablePlugins(NoPublishPlugin)
   .settings(
     description := "Integration Tests for the Kinesis Client"
   )
+  .jvmPlatform(allScalaVersions)
   .enableIntegrationTests
   .dependsOn(
     `kinesis-client-localstack` % IT,
     `kinesis-client-logging-circe` % IT
   )
 
-lazy val docs = project
+lazy val `smithy4s-client` = projectMatrix
+  .enablePlugins(Smithy4sCodegenPlugin)
+  .settings(
+    description := "Cats tooling for the Smithy4s Kinesis Client",
+    libraryDependencies ++= Seq(
+      S4S.http4sAws(smithy4sVersion.value),
+      Log4Cats.slf4j,
+      Smithy.rulesEngine % Smithy4s
+    ),
+    Compile / smithy4sAllowedNamespaces := List(
+      "smithy.rules",
+      "com.amazonaws.kinesis"
+    )
+  )
+  .jvmPlatform(last2ScalaVersions)
```

Comment:
Brought in sbt-projectmatrix to work through having a subset of scala versions supported here (Smithy4s AWS does not have a 2.12 offering)